packages/repository/src/model.ts

```diff
@@ -329,9 +330,9 @@ export abstract class Model {
 
     const json: AnyObject = {};
     const hiddenProperties: string[] = def.settings.hiddenProperties || [];
-    for (const p in def.properties) {
-      if (p in this && !hiddenProperties.includes(p)) {
-        copyPropertyAsJson(p);
+    for (const [pKey, pDef] of Object.entries(def.properties)) {
```

Comment:
Perhaps `propName` and `propDef` would be better names? In `def.properties` map, entries are keyed using the property name.

```suggestion
    for (const [propName, propDef] of Object.entries(def.properties)) {
```